### ISSUE TITLE
Issue with mat4.translate .

### ISSUE THREAD

**Issue by srikrishna0130** [toji/gl-matrix#344]
I think the mat4.translate function dosen't work as intended.
I performed the following operations :
`mat4.translate(modelviewmatrix,modelviewmatrix,translate);`
and,
` mat4.scale(modelviewmatrix,modelviewmatrix,scale);
  mat4.translate(modelviewmatrix,modelviewmatrix,translate);`

To achieve the translation observed in code 1 i needed to translate the second one by scale*translate
in the code 2.

I then had a look at the source code 
`   if (a === out) {

    out[12] = a[0] * x + a[4] * y + a[8] * z + a[12];

    out[13] = a[1] * x + a[5] * y + a[9] * z + a[13];

    out[14] = a[2] * x + a[6] * y + a[10] * z + a[14];

    out[15] = a[3] * x + a[7] * y + a[11] * z + a[15]; 

} `

To make the code work as intended i think this part needs a change to ,
`  if (a === out) {

    out[12] = x + a[12];

    out[13] = y + a[13];

    out[14] = z + a[14];

    out[15] = a[15];

  }` 

Sorry for any error i made...

**Comment by stefnotch** [comment]
Great question! I dusted off my rusty matrix mathematics and decided to take a stab at answering it.

Let's start off by making sure that we are on the same page here. A 4x4 transformation matrix looks like this:
![image](https://user-images.githubusercontent.com/10220080/48414910-348d3400-e74c-11e8-8ce6-b89b1250e0cc.png)


And, [the order of operations in a single matrix](https://github.com/toji/gl-matrix/blob/b482988ba75cd0b1996ea1f36692772a68d1b0b7/src/mat4.js#L1046) is the following in gl-matrix:
- Scale
- Rotation
- Translation

In other words, if you **translate** a matrix by a vector, gl-matrix **first** **scales** and **rotates** the vector. 

And this is exactly what is happening here
![image](https://user-images.githubusercontent.com/10220080/48415331-605ce980-e74d-11e8-8998-0bae36f267d1.png)


I hope that I explained it in an understandable manner. If you have any further questions, feel free to ask.

**Comment by stefnotch** [comment]
Issue closed because gl-matrix is doing the "right" thing, at least according to the usual WebGL/OpenGL conventions.
